src/runtime/storageNG/tests/store-test.ts

```diff
@@ -0,0 +1,231 @@
+    const activeStore = store.activate();
+
+    assert.equal(activeStore.constructor, DirectStore);
+  });
+
+  it('will propagate model updates from proxies to drivers', async () => {
+    DriverFactory.register(new MockStorageDriverProvider());
+
+    const store = new Store('string', Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+    const activeStore = store.activate();
+
+    const driver = activeStore['driver'] as MockDriver<CountData>;    
+    let capturedModel: CountData = null;
+    driver.send = async model => {capturedModel = model; return true;};
+
+    const count = new CRDTCount()
+    count.applyOperation({type: CountOpTypes.Increment, actor: 'me', version: {from: 0, to: 1}});
+
+    const result = await activeStore.onProxyMessage({type: ProxyMessageType.ModelUpdate, model: count.getData(), id: 1});
+    assert.isTrue(result);
+
+    assert.deepEqual(capturedModel, count.getData());
+  });
+
+  it('will apply and propagate operation updates from proxies to drivers', async() => {
+    DriverFactory.register(new MockStorageDriverProvider());
+
+    const store = new Store('string', Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+    const activeStore = store.activate();
+
+    const driver = activeStore['driver'] as MockDriver<CountData>;
+    let capturedModel: CountData = null;
+    driver.send = async model => {capturedModel = model; return true;};
+
+    const count = new CRDTCount();
+    const operation: CountOperation = {type: CountOpTypes.Increment, actor: 'me', version: {from: 0, to: 1}};
+
+    const result = await activeStore.onProxyMessage({type: ProxyMessageType.Operations, operations: [operation], id: 1});
+    assert.isTrue(result);
+
+    count.applyOperation(operation);
+     
+    assert.deepEqual(capturedModel, count.getData());
+  });
+
+  it('will respond to a model request from a proxy with a model', async () => {
+    DriverFactory.register(new MockStorageDriverProvider());
+
+    const store = new Store('string', Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+    const activeStore = store.activate();
+
+    const driver = activeStore['driver'] as MockDriver<CountData>;
+    driver.send = async model => true;
+
+    const count = new CRDTCount();
+    const operation: CountOperation = {type: CountOpTypes.Increment, actor: 'me', version: {from: 0, to: 1}};
+    count.applyOperation(operation);
+
+    let sentSyncRequest = false;
+
+    return new Promise(async (resolve, reject) => {
+      const id = activeStore.on(proxyMessage => {
+        if (proxyMessage.type === ProxyMessageType.Operations) {
+          assert.isFalse(sentSyncRequest);
+          sentSyncRequest = true;
+          activeStore.onProxyMessage({type: ProxyMessageType.SyncRequest, id});
+          return true;
+        }
+        assert.isTrue(sentSyncRequest);
+        if (proxyMessage.type === ProxyMessageType.ModelUpdate) {
```

Comment:
Why is this required? Looks like this test only drives the ops proxy message type..

Reply:
The first if clause sends a sync request in response to operations being delivered. I wanted a non-empty model inside the store and it seemed nice to test this whole flow rather than just push one in.